<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="crop" default="2.0"/>
  <arg name="freeSpace" default="0.2  "/>
  <arg name="name" default="sparse_map"/>
  <node name="pose_chk" pkg="space_quantization" type="pose_chck" output="screen">
    <remap from="/localization_pose" to="/amcl_pose"/>
    <remap from="/cloud_in" to="/hsrb/head_rgbd_sensor/depth_registered/rectified_points"/>
    <remap from="/cloud_out" to="/hsrb/cloud"/>
    <param name="angular_update" value="45"/>
    <param name="linear_update" value="0.25"/>
  </node>


  <node name="preprocessor" pkg="space_quantization" type="cloud_preprocessor" output="screen">
    <remap from="/cloud" to="/hsrb/cloud"/>
    <remap from="/processed_cloud" to="/hsrb/processed_cloud"/>
    <param name="cropEnable" value="false" type="bool"/>
    <param name="cropDistance" value="$(arg crop)" type="double"/>
    <param name="voxelSize" value="0.01" type="double"/>
    <param name="voxEnable" value="true" type="bool"/>
    <param name="baseFrame" value="map" type="string"/>
  </node>

  <node name="segmenter" pkg="space_quantization" type="segmentation_node" args="" output="screen">
    <remap from="cloud" to="/hsrb/processed_cloud"/>
    <remap from="labeled_cloud" to="/hsrb/labeled_cloud"/>
    <remap from="codebook" to="/hsrb/codebook"/>
    <param name="nClusters" value="25" type="int"/>
    <param name="iterations" value="10" type="int"/>
    <!-- <param name="visualizeCentroids" value="True" type="bool"/> <param name="visualizeSegCloud" value="false" type="bool"/> <param name="freeThr" value="0.05" type="double"/> -->
  </node>

  <node name="adjGraph" pkg="space_quantization" type="adjacency_graph_node" output="screen">
    <remap from="codebook" to="/hsrb/codebook"/>
    <remap from="graph_marker" to="/hsrb/graph_marker"/>
    <param name="k_neighboors" value="6" type="int"/>
    <param name="max_dist" value="3" type="double"/>
    <param name="safety_height" value="1.5" type="double"/>
    <param name="safety_radius" value="0.4" type="double"/>
    <param name="connection_radius" value="0.5" type="double"/>
    <param name="free_thr" value="$(arg freeSpace)" type="double"/>
    <param name="graph_file" value="$(find space_quantization)/graphs/$(arg name).txt" type="str"/>
  </node>
</launch>
